File: python/hash_table/q219.py
"""
Given an array of integers and an integer k, find out whether there are two distinct indices i and j in the array such that nums[i] = nums[j] and the absolute difference between i and j is at most k.
"""
def containsNearbyDuplicate(self, nums, k):
    """
    :type nums: List[int]
    :type k: int
    :rtype: bool
    """
    dic = {}
    for i,val in enumerate(nums):
        if val in dic:
            if i - dic[val] <= k:
                return True
            else:
                dic[val] = i
        else:
            dic[val] = i

    return False

File: c++/hash_table/q219.cpp
bool containsNearbyDuplicate(vector<int>& nums, int k) {
    unordered_map<int,int> um;
    for(int i = 0; i < nums.size(); i++){
        unordered_map<int,int>::iterator it = um.find(nums[i]);
        if(it == um.end())
            um[nums[i]] = i;
        else{
            if(i-it->second<=k)
                return true;
            it->second = i;
        }
    }
    
    return false;
}
